File: src/main/java/com/java/learnspark/dataanalysis/chapter01/TestSparkContext.java
package com.java.learnspark.dataanalysis.chapter01;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

import java.util.Arrays;
import java.util.List;

public class TestSparkContext {

    public static void main(String[] args) {
        SparkConf conf = new SparkConf().setMaster("local").setAppName("My App");
        JavaSparkContext sc = new JavaSparkContext(conf);
        System.out.println("java init:" + sc);

        List<Integer> data = Arrays.asList(1,2,3,4,5);
        JavaRDD<Integer> distData = sc.parallelize(data);
        int result = distData.reduce(Integer::sum);
        System.out.println("result: " + result);

    }
}

File: src/main/scala/com/scala/learnspark/dataanalysis/chapter01/TestSparkContext.scala
package com.scala.learnspark.dataanalysis.chapter01

import org.apache.spark.{SparkConf, SparkContext}

object TestSparkContext {

  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setMaster("local").setAppName("My App")
    val sc = new SparkContext(conf)
    println("scala init:" + sc)

    val data = Array(1,2,3,4,5)
    val distData = sc.parallelize(data)
    val result = distData.reduce(_ + _)
    println("result:" + result)
  }
}
